src/actions/index.js

```diff
@@ -0,0 +1,287 @@
+    clientId : CLIENT_ID,
+    redirectUri: REDIRECT_URI
+  });
+
+  return {
+    type: CREATE_AUTH,
+    payload: request
+  };
+}
+
+// LOGIN ACTIONS
+
+/**
+ *  Uses the Okta AuthSDK to establish a session given
+ *  "username" and "password" 
+ */
+export function loginAuth(user, auth) {
+  return function(dispatch) {
+    auth.signIn({
+      username: user.username,
+      password: user.password
+    })
+    .then((response) => {
+      dispatch(loginAuthSuccess(response));
+    })
+    .catch((error) => {
+      dispatch(loginAuthError(error));
+    });
+  };
+}
+
+function loginAuthSuccess(response) {
+  if(response.status === 'SUCCESS') {
+    return {
+      type: LOGIN_AUTH_SUCCESS,
+      payload: response
+    };  
+  }
+
+  return {
+    type: LOGIN_AUTH_ERROR,
+    payload: response
+  };
+}
+
+function loginAuthError(error) {
+  return {
+    type: LOGIN_AUTH_ERROR,
+    payload: error
+  };
+}
+
+// GET TOKENS ACTIONS
+
+/**
+ *  Given a sessionToken, returns "idToken" and/or "accessToken",
+ *  and user "clams"
+ */
+export function getTokens(login, auth) {
+  return function(dispatch) {
+    if(auth.session.exists()) {
+      auth.idToken.authorize({
+        sessionToken: login.res.sessionToken,
+        responseType: RESPONSE_TYPE,
+        scope: ['openid', 'email', 'profile', 'groups']
+      })
+      .then((response) => {
+        dispatch(getTokensSuccess(response));
+      })
+      .catch((error) => {
+        dispatch(getTokenError(error));
+      });
+    } else {
+      dispatch(getTokenError('session not exists'));
+    }
+  }
+}
+
+function getTokensSuccess(response) {
+  return {
+    type: GET_TOKEN_SUCCESS,
+    payload: response
+  };
+}
+
+function getTokenError(error) {
+  return {
+    type: GET_TOKEN_ERROR,
+    payload: error
+  };
+}
+
+// REFRESH TOKEN ACTIONS
+
+/**
+ *  Renews the current ID token
+ */
+export function renewIdToken(auth) {
+  return function(dispatch) {
+    if(auth.session.exists()) {
+      auth.idToken.refresh({ 'scope': ['openid', 'email', 'profile', 'groups'] })
+      .then((response) => {
+        dispatch(renewIdTokensSuccess(response));
+      })
+      .catch((error) => {
+        dispatch(renewIdTokenError(error));
+      });
+    } else {
+      dispatch(renewIdTokenError('session not exists'));
+    }
+  }
+}
+
+function renewIdTokensSuccess(response) {
+  return {
+    type: RENEW_ID_TOKEN_SUCCESS,
+    payload: response
+  };
+}
+
+function renewIdTokenError(error) {
+  return {
+    type: RENEW_ID_TOKEN_ERROR,
+    payload: error
+  };
+}
+
+// DECODE TOKEN ACTIONS
+
+/**
+ *  Given an "idToken", it decodes the
+ *  header, payload, and signiture
+ */
+export function decodeIdToken(token, auth) {
+  const decoded = auth.idToken.decode(token.idToken);
+  if(decoded) {
+    return {
+      type: DECODE_ID_TOKEN_SUCCESS,
+      payload: decoded
+    }
+  }
+
+  return {
+    type: DECODE_ID_TOKEN_ERROR,
+    payload: error
+  }
+}
+
+// REFRESH SESSION ACTIONS
```

Comment:
It might make sense to break out these action creator functions into separate files depending on the use case - for example the difference between the idToken flow vs the session flow (I haven't pulled down this code yet to see the examples, but I'm assuming they're different flows?)